src/pybind/kaldi_pybind.cc

```diff
@@ -18,13 +18,44 @@
 // See the Apache 2 License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 namespace py = pybind11;
 #include "matrix/kaldi-matrix.h"
+#include "util/table-types.h"
+#include "util/kaldi-table-inl.h"
 
 using namespace kaldi;
 
+template<class Holder>
+void sequential_matrix_reader(py::module &m, std::string suffix) {
```

Comment:
Is this a trick that is kosher / allowed by the pybind11 guys?  Looks dangerous.

Reply:
It works. There is really not much material about template class in their docs. Some people do it this way. What's the difference between this and using macro?

Reply:
Issues arise such as: what if the name-mangling is different because it's in function scope?  What if the compiler notices that the type is not used inside the function and optimizes it out, e.g. at higher optimization level?
But we can always change it to a macro later if this happens; I'll merge.

It would be nice to make this so you can do:
   `for key,value in Reader:`
but it might not be super trivial to do.  Would probably be easiest to accomplish at the Python-package level via a subclass.  (We'll eventually have a Python package that provides python-level wrapping around these objects where necessary.)